extensions/control-plane/provision/provision-bigquery/src/main/java/org/eclipse/edc/connector/provision/gcp/BigQueryProvisioner.java

```diff
@@ -0,0 +1,196 @@
+import org.eclipse.edc.gcp.iam.IamService;
+import org.eclipse.edc.gcp.iam.IamServiceImpl;
+import org.eclipse.edc.policy.model.Policy;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.response.ResponseStatus;
+import org.eclipse.edc.spi.response.StatusResult;
+import org.eclipse.edc.spi.types.TypeManager;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+public class BigQueryProvisioner implements Provisioner<BigQueryResourceDefinition, BigQueryProvisionedResource> {
+    private final GcpConfiguration gcpConfiguration;
+    private final Monitor monitor;
+    private final TypeManager typeManager;
+    private BigQueryProvisionService bqProvisionService;
+
+    private BigQueryProvisioner(GcpConfiguration gcpConfiguration, Monitor monitor, TypeManager typeManager) {
+        this.monitor = monitor;
+        this.gcpConfiguration = gcpConfiguration;
+        this.typeManager = typeManager;
+    }
+
+    @Override
+    public boolean canProvision(ResourceDefinition resourceDefinition) {
+        return resourceDefinition instanceof BigQueryResourceDefinition;
+    }
+
+    @Override
+    public boolean canDeprovision(ProvisionedResource resourceDefinition) {
+        return resourceDefinition instanceof BigQueryProvisionedResource;
+    }
+
+    @Override
+    public CompletableFuture<StatusResult<ProvisionResponse>> provision(
+            BigQueryResourceDefinition resourceDefinition, Policy policy) {
+        var target = getTarget(resourceDefinition);
+        monitor.info("BigQuery Provisioner provision " + target.getTableName());
+        // TODO check if injected IAM service can be used (ADC / refresher).
+        var bqIamService = getIamService(resourceDefinition);
+        var serviceAccountName = getServiceAccountName(resourceDefinition);
+
+        var tableName = Optional.ofNullable(target.table())
+                .orElseGet(() -> {
+                    var generatedTableName = resourceDefinition.getId();
+                    monitor.debug("BigQuery Provisioner table name generated: " + generatedTableName);
+                    return generatedTableName;
+                });
+        var resourceName = tableName + "-table";
+
+        // TODO update target with the generated table name.
+
+        try {
+            String serviceAccountEmail = null;
+            GcpServiceAccount serviceAccount = null;
+            if (serviceAccountName != null) {
+                serviceAccount = bqIamService.getServiceAccount(serviceAccountName);
+                serviceAccountEmail = serviceAccount.getEmail();
+            }
+
+            if (bqProvisionService == null) {
+                bqProvisionService = BigQueryProvisionServiceImpl.Builder.newInstance(gcpConfiguration, target, monitor)
+                        .serviceAccount(serviceAccountEmail)
+                        .build();
+            }
```

Comment:
as I said in the first review, this behavior is dangerous because the client will contain information from the first call, so it's not thread-safe.

My suggestion would be to make it state-less, passing the `target` as parameter of the `tableExists` method, so you could keep the `BigQueryProvisionService` instance as immutable field of the Provisioner class and thread-safe.

Reply:
I will restore the target parameter, I agree it is in general a good idea.
BigQueryProvisioner now uses a different service though, after the split: extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/bigquery/service/BigQueryProvisionServiceImpl.java
The new service doesn't use threads. 

I will remove from the PR the other two services, for the source and the sink, since they will be used in the data plane.

Reply:
> BigQueryProvisioner now uses a different service though, after the split: extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/bigquery/service/BigQueryProvisionServiceImpl.java The new service doesn't use threads.

I didn't refer to threads, but to thread-safeness. 
Generally speaking we'd want a service to be thread safe, so that it don't provide side effects when called in a multi-threaded environment (like this). To make a service thread-safe, it must be stateless.

Reply:
I remove the member variable for the service, the reason for using it was to have a mock in place for the test.

Reply:
Done